[IMP] general: companies rewrite (#8470)
odoo/documentation: content/applications/general/companies.rst

Companies
=========

A centralized management environment allows an administrator to select multiple companies
simultaneously, and set their specific warehouses, customers, equipment, and contacts. It provides
the ability to generate reports of aggregated figures without switching interfaces, which
facilitates daily tasks, and enhances the overall management process.

.. danger::
   Utilizing a multi-company environment in an Odoo database warrants a subscription change from
   *Standard* (or *One App Free*) to a *Custom* plan. Should the administrator not migrate the
   subscription to a *Custom* plan after adding another company, the database risks being
   deactivated. For more information, consult the `Odoo Pricing page
   <https://www.odoo.com/pricing-plan>`_.

   If the customer is on the *Standard* plan, and adds multi-company functionality to the database,
   it triggers an upsell for the *Custom* plan.

>    If they are on a *Yearly* or *Multi-year* contract, this creates an upsell order, with a 30 day

Reviewer: ```suggestion
   If they are on a *Yearly* or *Multi-year* contract, this creates an upsell order, with a 30-day
```

   limit (risking deactivation). If multi-company functionality is fully removed within that time
   period, the next time the database pings, the upsell order and limit are removed automatically.

   If the customer is on a *Monthly* contract, once the next bill is created, and multi-company

>    functionality is still detected while they are on *Standard*, they are automatically switched,

Reviewer: ```suggestion
   functionality is still detected while they are on a *Standard* plan, they are automatically switched,
```

   and billed on the *Custom* plan.

   Should the customer want to go back to *Standard*, the Customer Success Team **must** be
   contacted.

To create a new company, navigate to :menuselection:`Settings app --> Companies section`, and click
:guilabel:`Manage Companies`. Then, click :guilabel:`New` to create a new company.

Proceed to fill out the new company form that appears.

.. tip::
   To archive a company, navigate to :menuselection:`Settings app --> Companies section --> Manage
   Companies`. Then, tick the checkbox to the left of the company to be archived. If the

>    :guilabel:`Companies` page is not list view, click the :guilabel:`≣ (four bars)` icon, located in

Reviewer: ```suggestion
   :guilabel:`Companies` page is not in list view, click the :guilabel:`≣ (four bars)` icon, located in
```

   in the top-right corner of the page.

>    After selecting the appropriate company, click the :guilabel:`Action ⚙️ (gear)` icon, and select

Reviewer 2: On the UI the button is labeled "⚙️ Actions" and not "Action ⚙️", no need for icon descriptor when an icon is accompanied by a label :)
```suggestion
   After selecting the appropriate company, click the :guilabel:`⚙️ Actions` icon, and select
```

   :guilabel:`Archive` from the resulting drop-down menu.

   To ensure all records related to the archived company are archived, contact Odoo's `Support Team
   <https://www.odoo.com/help>`_.

   Should a record not be archived, there is a risk of reactivating the archived company, and
   creating the upsell again.

.. _companies/manage:

Manage companies and records
============================

Go to :menuselection:`Settings app --> Companies section --> Manage Companies`. Then, either click
:guilabel:`New`, and fill in the form with the company's information, or select a pre-existing
company to edit it.

.. image:: companies/company-info.png
   :align: center
   :alt: Overview of a new company's form in Odoo.

.. tip::

>    Activate the :ref:`developer mode <developer-mode>` to set *Social Media* accounts and

Reviewer 2: No need to format _social media_ IMO
```suggestion
   Activate the :ref:`developer mode <developer-mode>` to set social media accounts and
```

   company-specific email system parameters. See this documentation on
   :doc:`../marketing/social_marketing` and :ref:`email_servers/outbound`.

>    Companies also have a :guilabel:`Parent Company` set on the *company* form in :ref:`developer

Reviewer: ```suggestion
   Companies also have a :guilabel:`Parent Company` set on the company form in :ref:`developer
```

>    Companies also have a :guilabel:`Parent Company` set on the company form in :ref:`developer
>    mode <developer-mode>`.

Reviewer 2: early break
```suggestion
   Companies also have a :guilabel:`Parent Company` set on the company form in :ref:`developer mode
   <developer-mode>`.
```

Switch between companies
------------------------

Switch between (or select) multiple companies, by clicking on the company name, located in the
far-right corner of the header menu, anywhere throughout the database. Tick the checkboxes next to
the desired company name(s) to activate them. The highlighted company represents the current
environment that is in use. To switch environments, click on the desired company name.

.. example::
   In the example below, the user has access to eight companies, two are activated, and the

>    environment in belongs to: *My Company (San Francisco)*.

Reviewer: ```suggestion
   environment the database is in belongs to: *My Company (San Francisco)*.
```

   .. image:: companies/multi_companies_menu_dashboard.png
      :align: center
      :alt: View of the companies menu through the main dashboard in Odoo.

Share records
-------------

Data (such as, products, contacts, and equipment) can be shared, or set to be shown for a specific
company only. To do so, on their forms, choose between:

- *A blank field*: the record is shared within all companies.
- *Adding a company*: the record is visible to users logged in to that specific company.

.. image:: companies/product_form_company.png
   :align: center
   :alt: View of a product's form emphasizing the company field in Odoo Sales.

> When an environment is selected from the top menu along with an additional company, records are

Reviewer: ```suggestion
When an environment is selected from the top menu, along with an additional company, records are
```

shared between the two companies.

Branches
========

Branches are available to add to a company. Branches can be added by navigating to
:menuselection:`Settings app --> Companies section --> Manage Companies`. Then, select the desired
company from the list. From the company detail form, open the :guilabel:`Branches` tab. To add a
branch, click :guilabel:`Add a line`, and fill out the :guilabel:`Create Branches` pop-up form that
appears.

.. image:: companies/add-branch.png
   :align: center
   :alt: Add a branch to a company with branches and add a line highlighted.

.. tip::

>    Activate the :ref:`developer mode <developer-mode>` to set *Social Media* accounts and

Reviewer 2: ```suggestion
   Activate the :ref:`developer mode <developer-mode>` to set social media accounts and
```

   company-specific email system parameters. See this documentation on
   :doc:`../marketing/social_marketing` and :ref:`email_servers/outbound`.

>    Branches also have a :guilabel:`Parent Company` set on the *branch* form in :ref:`developer mode

Reviewer: ```suggestion
   Branches also have a :guilabel:`Parent Company` set on the branch form in :ref:`developer mode
```

   <developer-mode>`. Accounting and fiscal localizations for the branch are set on the
   :guilabel:`Parent Company`. To do so, select the company from the *company selector* in the top

>    menu and go to :menuselection:`Settings app --> Accounting --> Fiscal Localization`.

Reviewer: ```suggestion
   menu, and go to :menuselection:`Settings app --> Accounting --> Fiscal Localization`.
```

.. danger::
   Adding a branch to a company triggers an upsell, should the database be in the *Standard* or
   *One App Free* pricing plans. Adding one or more branches constitutes a multi-company database,
   and needs to be moved to the *Custom* pricing plan.

   For more information on pricing, see `Odoo's pricing <https://www.odoo.com/pricing-plan>`_ page.

> Employees access

Reviewer: ```suggestion
Employee access
```

================

Once companies are created, manage the employees' :doc:`Access Rights <users/access_rights>` for
*Multi Companies*.

To access the *Access Rights*, navigate to :menuselection:`Settings app --> Users section --> Manage
Users`.

From the :guilabel:`Users` page, select a user from the list to modify. Then, either change the
fields for :guilabel:`Allowed Companies` or :guilabel:`Default Company`.

Multiple companies can be set for :guilabel:`Allowed Companies`, and *only one* can be set as the
:guilabel:`Default Company`.

.. image:: companies/access_rights_multi_companies.png
   :align: center
   :alt: View of an user form emphasizing the multi companies field under the access rights tabs
         in Odoo.

> If an administrator has multiple companies *activated* on the database, and is editing a record, the

Reviewer: ```suggestion
If an administrator has multiple companies activated on the database, and is editing a record, the
```

editing occurs on the record's related company.

.. example::
   If editing a sale order issued under `JS Store US`, while working on the `JS Store Belgium`
   environment, the changes are applied under `JS Store US` (the company from which the sale order
   was issued).

When creating a record, the company taken into account is:

> - The current company selected in the company selector in the upper-right hand of the screen (the

Reviewer: ```suggestion
- The current company selected in the company selector, in the upper-right hand of the screen (the
```

  one that is highlighted/active)

**OR**

- No company is set (because none is set on the product and contact forms, for example)

**OR**

> - The company set is the company linked to the document (the same as if a record is being edited).

Reviewer: ```suggestion
- The company set is the company linked to the document (the same as if a record is being edited)
```

Document format
===============

> To set documents formats according to each company, *activate* and *select* the respective company,

Reviewer: ```suggestion
To set document formats according to each company, activate and select the respective company,
```

and, under the :menuselection:`Settings app`, click on :guilabel:`Configure Document Layout` and

> edit the info as needed.

Reviewer: ```suggestion
edit the information, as needed.
```

> and, under the :menuselection:`Settings app`, click on :guilabel:`Configure Document Layout` and
> edit the information as needed.

Reviewer 2: Suggestion to stay consistent with the navigation path throughout the doc :)
```suggestion
and, under the :menuselection:`Settings app --> Companies section`, click on :guilabel:`Configure
Document Layout` and edit the information as needed.
```

.. image:: companies/document_layout.png
   :align: center
   :alt: View of the settings page emphasizing the document layout field in Odoo.

> :guilabel:`Company Details` can be edited on the document layout. By default this field is populated

Reviewer: ```suggestion
:guilabel:`Company Details` can be edited on the document layout. By default, this field is populated
```

> from the company info listed when navigating here: :menuselection:`Settings app --> Companies -->

Reviewer: ```suggestion
from the company information listed, when navigating here: :menuselection:`Settings app --> Companies section -->
```

--> Companies section --> Manage Companies`, and select a company from the list.

> Inter-Company Transactions

Reviewer 2: Sentence case header please :)
```suggestion
Inter-company transactions
```

Author: This was a change from KC.

==========================

> **First**, activate the :ref:`developer mode <developer-mode>`. Then, make sure each one of the

Reviewer 2: The bold **First** formatting seems a little strange to me, but I understand it is calling attention that debug mode **must** be activated in order to see the _Inter-Company Transactions_ tab. 

Maybe there is a better way to structure this notice just before navigating to the _Inter-Company Transactions_ tab since debug mode doesnt necessarily need to be enabled to configure the Chart of Accounts, Taxes, etc...?
```suggestion
First, activate the :ref:`developer mode <developer-mode>`. Then, make sure each one of the
```

shared between the two companies.

Branches
========

Branches are available to add onto a company. Branches can be added by navigating to
:menuselection:`Settings app --> Manage Companies` and click on the :guilabel:`Branches` tab. To add
a branch click :guilabel:`Add a line` and fill out the form.

.. image:: companies/add-branch.png
   :align: center
   :alt: Add a branch to a company with branches and add a line highlighted.

Branches have a :guilabel:`Parent Company` set on the *branch* form. Accounting and fiscal
localizations for the branch are set on the :guilabel:`Parent Company`. To do so, select the company
from the *company selector* in the top menu and go to :menuselection:`Settings app --> Accounting
--> Fiscal Localization`.

.. warning::
   Adding a branch to a company will trigger an upsell, should the database be in the *Standard* or
   *One App Free* pricing plans. Adding one or more branches constitutes as a multi-company database
   and will need to be moved to the *Custom* pricing plan. For more information on pricing, see
   `Odoo's pricing <https://www.odoo.com/pricing-plan>`_ page.

Employees access
================

> Once companies are created, manage the employees' :doc:`Access Rights <users/access_rights>`

Reviewer 3: Early line break, less than 100 characters

for *Multi Companies*. To access the *Access Rights* navigate to :menuselection:`Settings app -->
Manage Users`. Either change the fields for :guilabel:`Allowed Companies` or :guilabel:`Default

> Company`. Multiple companies can be set for :guilabel:`Allowed Companies` and only one can be set
> for the :guilabel:`Default Company`.

Reviewer 3: ```suggestion
Company`. Multiple companies can be set for :guilabel:`Allowed Companies`, and only one can be set
as the :guilabel:`Default Company`.
```

.. image:: companies/product_form_company.png
   :align: center
   :alt: View of a product's form emphasizing the company field in Odoo Sales.

When an environment is selected from the top menu along with an additional company, records are
shared between the two companies.

Branches
========

Branches are available to add to a company. Branches can be added by navigating to
:menuselection:`Settings app --> Manage Companies` and click on the :guilabel:`Branches` tab. To add
a branch, click :guilabel:`Add a line` and fill out the form.

.. image:: companies/add-branch.png
   :align: center
   :alt: Add a branch to a company with branches and add a line highlighted.

Branches have a :guilabel:`Parent Company` set on the *branch* form. Accounting and fiscal
localizations for the branch are set on the :guilabel:`Parent Company`. To do so, select the company
from the *company selector* in the top menu and go to :menuselection:`Settings app --> Accounting

> --> Fiscal Localization`.

Author: I've added clarification. 

Thanks for spotting this.

   :align: center
   :alt: View of an user form emphasizing the multi companies field under the access rights tabs
         in Odoo.

If an administrator has multiple companies *activated* on the database, and is **editing** a record,
the editing occurs on the record's related company.

.. example::
   If editing a sale order issued under `JS Store US` while working on the `JS Store Belgium`
   environment, the changes are applied under `JS Store US` (the company from which the sale order
   was issued).

When **creating** a record, the company taken into account is:

- The current company (the one active) or,
- No company is set (on products and contacts' forms for example) or,
- The company set is the one linked to the document (the same as if a record is being edited).

Document format
===============

To set documents' formats according to each company, *activate* and *select* the respective company,
and, under :menuselection:`Settings app`, click on :guilabel:`Configure Document Layout` under the
:guilabel:`Companies` section.

.. image:: companies/document_layout.png
   :align: center
   :alt: View of the settings page emphasizing the document layout field in Odoo.

> 

Author: I've added in clarification and some more details.

- :doc:`Fiscal Localizations <../finance/fiscal_localizations>`
- :doc:`Pricelists <../sales/sales/products_prices/prices/pricing>`

Now, navigate to the :guilabel:`Inter-Company Transactions` tab under the individual company's
settings. To do so, go to: :menuselection:`Settings app --> Manage Companies` and select a company.

> With the respective company *activated* and *selected*, choose whether operations between

Reviewer 3: Early line break, less than 100 characters

  the selected company warehouse when a sales/purchase order is confirmed for the selected company.
  If instead of a drafted purchase/sales order, it should be validated, then, enable
  :guilabel:`Automatic Validation`.

.. image:: companies/inter_company_transactions.png
   :align: center
   :alt: View of the settings page emphasizing the inter company transaction field in Odoo.

.. example::
   :guilabel:`Synchronize invoice/bills`: an invoice posted on `JS Store Belgium`, for `JS Store
   US`, automatically creates a vendor bill generates a drafted purchase/sales order using the
   selected company warehouse when a sales/purchase order is confirmed for the selected company. If
   instead of a drafted purchase/sales order, it should be validated, then, enable
   :guilabel:`Automatic Validation`.

.. example::
   :guilabel:`Synchronize sales/purchase order`: when a sale order for `JS Store US` is confirmed on
   `JS Store Belgium`, a purchase order on `JS Store Belgium` is automatically created (and
   confirmed if the :guilabel:`Automatic Validation` feature was enabled).

.. note::
   Products have to be configured as :guilabel:`Can be sold` and must be shared between the

>    companies.

Author: No this is a general note

  warehouse when a sales order is confirmed for the selected company. If instead of a drafted sales
  order, it should be validated, then, enable :guilabel:`Automatic Validation`.
- :guilabel:`Synchronize Purchase Order`: generates a drafted purchase order using the selected
  company warehouse when a purchase order is confirmed for the selected company. If instead of a
  drafted purchase order, it should be validated, then, enable :guilabel:`Automatic Validation`.
- :guilabel:`Synchronize Sales and Purchase Order`: generates a drafted purchase/sales order using
  the selected company warehouse when a sales/purchase order is confirmed for the selected company.
  If instead of a drafted purchase/sales order, it should be validated, then, enable
  :guilabel:`Automatic Validation`.

.. image:: companies/inter_company_transactions.png
   :align: center
   :alt: View of the settings page emphasizing the inter company transaction field in Odoo.

.. note::
   Products **must** be configured as :guilabel:`Can be sold` and be shared between the companies.

.. example::
   :guilabel:`Synchronize invoice/bills`: an invoice posted on `JS Store Belgium`, for `JS Store

>    US`, automatically creates a vendor bill generates a drafted purchase/sales order using the

Reviewer: ```suggestion
   US`, automatically creates a vendor bill, and generates a drafted purchase/sales order using the
```

>    selected company warehouse when a sales/purchase order is confirmed for the selected company. If

Reviewer: ```suggestion
   selected company warehouse, when a sales/purchase order is confirmed for the selected company. If,
```

>    instead of a drafted purchase/sales order, it should be validated, then, enable
>    :guilabel:`Automatic Validation`.

Reviewer: ```suggestion
   instead of a drafted purchase/sales order, it should be validated, enable
   :guilabel:`Automatic Validation`.
```

- :doc:`Fiscal Localizations <../finance/fiscal_localizations>`
- :doc:`Pricelists <../sales/sales/products_prices/prices/pricing>`

> Next, navigate to the :guilabel:`Inter-Company Transactions` tab, on the individual company's
> detail form. To do so, go to the :menuselection:`Settings app --> Companies section --> Manage
> Companies`, and select a company from the list.

Reviewer 2: Suggestion to move the navigation instructions before the tab configuration, as this is a more logical flow when reading through.
```suggestion
Next, navigate to :menuselection:`Settings app --> Companies section --> Manage Companies`. Then,
select the desired company from the list. On the company form, select the :guilabel:`Inter-Company
Transactions` tab, on the individual company's detail form.
```

> With the respective company activated and selected, choose one of the following options:

Reviewer 2: Add the field name for specificity 
```suggestion
With the respective company activated and selected, choose one of the following :guilabel:`Rule`
options:
```

> With the respective company *activated* and *selected*, choose one of the following options:

Reviewer: ```suggestion
With the respective company activated and selected, choose one of the following options:
```

> - :guilabel:`Do not synchronize`: don't synchronize any inter-company transactions.

Reviewer: ```suggestion
- :guilabel:`Do not synchronize`: do not synchronize any inter-company transactions.
```

- :guilabel:`Synchronized invoice/bills`: generates a bill/invoice when a company confirms a
  bill/invoice for the selected company.
- :guilabel:`Synchronize Sales Order`: generates a drafted sales order using the selected company
  warehouse, when a sales order is confirmed for the selected company. If, instead of a drafted

>   sales order, it should be validated, enable :guilabel:`Automatic Validation`.*

Reviewer 2: Add a space and escape symbol 
```suggestion
  sales order, it should be validated, enable :guilabel:`Automatic Validation`. \*
```

- :guilabel:`Synchronize Purchase Order`: generates a drafted purchase order using the selected
  company warehouse, when a purchase order is confirmed for the selected company. If, instead of a

>   drafted purchase order, it should be validated, enable :guilabel:`Automatic Validation`.*

Reviewer 2: ```suggestion
  drafted purchase order, it should be validated, enable :guilabel:`Automatic Validation`. \*
```

- :guilabel:`Synchronize Sales and Purchase Order`: generates a drafted purchase/sales order using
  the selected company warehouse, when a sales/purchase order is confirmed for the selected company.
  If, instead of a drafted purchase/sales order, it should be validated, enable :guilabel:`Automatic

>   Validation`.*

Reviewer 2: ```suggestion
  Validation`. \*
```

>   \*The given option needs to be selected so that :guilabel:`Automatic Validation` appears in the

Reviewer: i'm assuming the / is make sure the asterisk appears as a footnote, and not just an italicized sentence, correct?

Author: Yes that is correct on the asterisk

>   \*The given option needs to be selected so that :guilabel:`Automatic Validation` appears in the

Reviewer: ```suggestion
  \*The given option needs to be selected, so :guilabel:`Automatic Validation` appears in the
```

>   \*The given option needs to be selected, so :guilabel:`Automatic Validation` appears in the
>   configuration.

Reviewer 2: If this notice applies to all the * points, lets break it out from the last bullet point (also add a space).
```suggestion
\* The given option needs to be selected, so :guilabel:`Automatic Validation` appears in the
configuration.
```

.. image:: companies/inter_company_transactions.png
   :align: center
   :alt: View of the settings page emphasizing the inter company transaction field in Odoo.

.. note::
   Products **must** be configured as :guilabel:`Can be sold` and shared between the companies. See
   :doc:`../inventory_and_mrp/inventory/product_management/product_tracking/type`.

.. example::
   :guilabel:`Synchronize invoice/bills`: an invoice posted on `JS Store Belgium`, for `JS Store
   US`, automatically creates a vendor bill, and generates a drafted purchase/sales order using the

>    selected company warehouse when a sales/purchase order is confirmed for the selected company. If

Reviewer: ```suggestion
   selected company warehouse, when a sales/purchase order is confirmed for the selected company. If,
```

>    instead of a drafted purchase/sales order, it should be validated, then, enable

Reviewer: ```suggestion
   instead of a drafted purchase/sales order, it should be validated, enable
```

   :guilabel:`Automatic Validation`.

   :guilabel:`Synchronize sales/purchase order`: when a sale order for `JS Store US` is confirmed on
   `JS Store Belgium`, a purchase order on `JS Store Belgium` is automatically created (and

>    confirmed if the :guilabel:`Automatic Validation` feature was enabled).

Reviewer: ```suggestion
   confirmed, if the :guilabel:`Automatic Validation` feature was enabled).
```

.. tip::
   Remember to test all workflows as a user *other* than the administrator.

.. seealso::

>    - :doc:`Multi-company Guidelines </developer/howtos/company>`

Reviewer 2: Update to relative path
```suggestion
   - :doc:`Multi-company Guidelines <../../developer/howtos/company>`
```
